src/Illuminate/Support/Collection.php

```diff
@@ -768,6 +768,16 @@ public function toJson($options = 0)
 	}
 
 	/**
+	 * Get an hash of the collection.
+	 *
+	 * @return string
+	 */
+	public function hash()
+	{
+		return md5($this);
```

Comment:
BTW, I prefer to implement a `Collection::equals(Collection $collection)` function to check equality instead of relaying on comparing hashes..